complex.c

```diff
@@ -560,10 +560,11 @@ nucomp_f_complex(int argc, VALUE *argv, VALUE klass)
     if (rb_scan_args(argc, argv, "11:", &a1, &a2, &opts) == 1) {
         a2 = Qundef;
     }
+    if (argc > 0 && CLASS_OF(a1) == rb_cComplex && a2 == Qundef) {
+        return a1;
+    }
```

Comment:
I think this should be after checking keywords by `rb_opts_exception_p`.

Reply:
Agreed, just pushed a new commit for that, and a test that checks that ArgumentError is raised for a bad keyword.